XiEditor/AppDelegate.swift

```diff
@@ -90,6 +94,20 @@ class AppDelegate: NSObject, NSApplicationDelegate {
         return nil
     }
 
+    /// Passed an NSFontManager instance (as on a user-initiated font change)
+    /// computes the next set of drawing metrics and updates cached styles.
+    func handleFontChange(fontManager: NSFontManager) {
+        let newFont = fontManager.convert(textMetrics.font)
+        textMetrics = TextDrawingMetrics(font: newFont)
+        styleMap.updateFont(to: newFont)
+
+        for doc in NSApplication.shared().orderedDocuments {
+            guard let doc = doc as? Document else { continue }
+            doc.editViewController?.updateGutterWidth()
+            doc.editViewController?.editView.needsDisplay = true
```

Comment:
oh, one last thing: we might want, on font change, to re send the `scroll` notification? Otherwise there may be visual jank when changing font sizes.